src/ServicePulse.Host/vue/src/components/monitoring/EndpointListRow.vue

```diff
@@ -1,70 +1,65 @@
 import { useFormatTime, useFormatLargeNumber } from "../../composables/formatter";
 import { smallGraphsMinimumYAxis } from "./formatGraph";
 import SmallGraph from "./SmallGraph.vue";
 import { useMonitoringHistoryPeriodStore } from "@/stores/MonitoringHistoryPeriodStore";
-import { useMonitoringStore } from "../../stores/MonitoringStore";
+import { useMonitoringStore, type MonitoringStore } from "../../stores/MonitoringStore";
 import { storeToRefs } from "pinia";
+import type { GroupedEndpoint, Endpoint, EndpointValues } from "@/resources/Endpoint";
 
-const settings = defineProps({
-  endpoint: Object,
-});
+const settings = defineProps<{
+  endpoint: GroupedEndpoint | Endpoint;
+}>();
 
 const monitoringHistoryPeriodStore = useMonitoringHistoryPeriodStore();
-const monitoringStore = useMonitoringStore();
-const isGrouped = computed(() => monitoringStore.endpointListIsGrouped);
-const endpoint = computed(() => {
-  return isGrouped.value ? settings.endpoint.endpoint : settings.endpoint;
+const monitoringStore: MonitoringStore = useMonitoringStore();
+const isGrouped = computed<boolean>(() => monitoringStore.endpointListIsGrouped);
+const endpoint = computed<Endpoint>(() => {
+  return isGrouped.value ? (settings.endpoint as GroupedEndpoint).endpoint : (settings.endpoint as Endpoint);
 });
 const shortName = computed(() => {
-  return isGrouped.value ? settings.endpoint.shortName : "";
+  return isGrouped.value ? (settings.endpoint as GroupedEndpoint).shortName : endpoint.value.name;
 });
 const router = useRouter();
 const supportsEndpointCount = ref();
 const { historyPeriod: selectedPeriod } = storeToRefs(monitoringHistoryPeriodStore);
 
-function navigateToEndpointDetails($event, endpointName) {
-  if ($event.target.localName !== "button") {
-    router.push({ name: "endpoint-details", params: { endpointName: endpointName }, query: { historyPeriod: selectedPeriod.value.pVal } });
-  }
+function navigateToEndpointDetails(endpointName: string) {
+  router.push({ name: "endpoint-details", params: { endpointName: endpointName }, query: { historyPeriod: selectedPeriod.value.pVal } });
 }
 
-function formatGraphDuration(input) {
+function formatGraphDuration(input: EndpointValues) {
   if (input) {
     const lastValue = input.points.length > 0 ? input.points[input.points.length - 1] : 0;
     return useFormatTime(lastValue);
   }
-  return input;
+  return { value: "0", unit: "" };
 }
-function formatGraphDecimal(input, deci) {
-  if (input) {
-    const lastValue = input.points.length > 0 ? input.points[input.points.length - 1] : 0;
-    let decimals = 0;
-    if (lastValue < 10 || input > 1000000) {
-      decimals = 2;
-    }
-    return useFormatLargeNumber(lastValue, deci || decimals);
-  } else {
-    input = {
-      points: [],
-      average: 0,
-      displayValue: 0,
-    };
-    return input;
+
+function formatGraphDecimal(input: EndpointValues, deci: number) {
+  input = input ?? {
```

Comment:
`input` is not nullable?
Should it be?

Reply:
removed this function in preference for the shared version